tests/Schema/ComparatorTest.php

```diff
@@ -619,44 +619,6 @@ public function testAutoIncrementNoSequences(): void
         self::assertCount(0, $diff->newSequences);
     }
 
-    /**
-     * You can get multiple drops for a FK when a table referenced by a foreign
-     * key is deleted, as this FK is referenced twice, once on the orphanedForeignKeys
-     * array because of the dropped table, and once on changedTables array. We
-     * now check that the key is present once.
-     */
-    public function testAvoidMultipleDropForeignKey(): void
-    {
-        $oldSchema = new Schema();
-
-        $tableA = $oldSchema->createTable('table_a');
-        $tableA->addColumn('id', 'integer');
-
-        $tableB = $oldSchema->createTable('table_b');
-        $tableB->addColumn('id', 'integer');
-
-        $tableC = $oldSchema->createTable('table_c');
-        $tableC->addColumn('id', 'integer');
-        $tableC->addColumn('table_a_id', 'integer');
-        $tableC->addColumn('table_b_id', 'integer');
-
-        $tableC->addForeignKeyConstraint($tableA->getName(), ['table_a_id'], ['id']);
-        $tableC->addForeignKeyConstraint($tableB->getName(), ['table_b_id'], ['id']);
-
-        $newSchema = new Schema();
-
-        $tableB = $newSchema->createTable('table_b');
-        $tableB->addColumn('id', 'integer');
-
-        $tableC = $newSchema->createTable('table_c');
-        $tableC->addColumn('id', 'integer');
-
-        $schemaDiff = $this->comparator->compareSchemas($oldSchema, $newSchema);
-
-        self::assertCount(1, $schemaDiff->changedTables['table_c']->getDroppedForeignKeys());
```

Comment:
This test is being removed as invalid. Consider instrumenting it with the following code:
```php
$platform = new MySQLPlatform();

echo '-- Schema', PHP_EOL;
foreach ($oldSchema->toSql($platform) as $statement) {
    echo $statement, ';', PHP_EOL;
}

echo '-- Diff', PHP_EOL;
foreach ($schemaDiff->toSql($platform) as $statement) {
    echo $statement, ';', PHP_EOL;
}
```
If run on `4.0.x`, it produces the following statements (the `PRIMARY KEY` clauses are not defined by the test schema and were added manually for the sake of correctness):
```sql
-- Schema
CREATE TABLE table_a (id INT NOT NULL PRIMARY KEY);
CREATE TABLE table_b (id INT NOT NULL PRIMARY KEY);
CREATE TABLE table_c (id INT NOT NULL, table_a_id INT NOT NULL, table_b_id INT NOT NULL, INDEX IDX_7365E39D71081393 (table_a_id), INDEX IDX_7365E39D63BDBC7D (table_b_id));
ALTER TABLE table_c ADD CONSTRAINT FK_7365E39D71081393 FOREIGN KEY (table_a_id) REFERENCES table_a (id);
ALTER TABLE table_c ADD CONSTRAINT FK_7365E39D63BDBC7D FOREIGN KEY (table_b_id) REFERENCES table_b (id);

-- Diff
ALTER TABLE table_a DROP FOREIGN KEY FK_7365E39D71081393;
DROP TABLE table_a;
ALTER TABLE table_c DROP FOREIGN KEY FK_7365E39D63BDBC7D;
DROP INDEX IDX_7365E39D71081393 ON table_c;
DROP INDEX IDX_7365E39D63BDBC7D ON table_c;
ALTER TABLE table_c DROP table_a_id, DROP table_b_id;
```
The assertion in question passes only because the `DROP FOREIGN KEY FK_7365E39D71081393` statement is mistakenly executed on `table_a` although it should be executed on `table_c`.

With the functionality of handling orphaned foreign keys removed, the test starts failing because now the SQL looks like this:
```sql
-- Schema
CREATE TABLE table_a (id INT NOT NULL PRIMARY KEY);
CREATE TABLE table_b (id INT NOT NULL PRIMARY KEY);
CREATE TABLE table_c (id INT NOT NULL, table_a_id INT NOT NULL, table_b_id INT NOT NULL, INDEX IDX_7365E39D71081393 (table_a_id), INDEX IDX_7365E39D63BDBC7D (table_b_id));
ALTER TABLE table_c ADD CONSTRAINT FK_7365E39D71081393 FOREIGN KEY (table_a_id) REFERENCES table_a (id);
ALTER TABLE table_c ADD CONSTRAINT FK_7365E39D63BDBC7D FOREIGN KEY (table_b_id) REFERENCES table_b (id);

-- Diff
DROP TABLE table_a;
ALTER TABLE table_c DROP FOREIGN KEY FK_7365E39D71081393;
ALTER TABLE table_c DROP FOREIGN KEY FK_7365E39D63BDBC7D;
DROP INDEX IDX_7365E39D71081393 ON table_c;
DROP INDEX IDX_7365E39D63BDBC7D ON table_c;
ALTER TABLE table_c DROP table_a_id, DROP table_b_id;
```
Both constraints are dropped on `table_c` (which is correct) but this is not what the assertion expects.

Note that the resulting SQL is still invalid because `table_a` cannot be dropped until it's referenced by a foreign key. The functionality of dropping multiple tables should be reworked accordingly but it seems to be out of the scope of this change since it didn't work properly even before this change.

Reply:
A test that was passing by accident… must have been quite confusing, thanks for the detailed explanation!